script.js

```diff
@@ -52,4 +52,147 @@ function valutaConverter(event) {
+
+  const selectedCurrency = brandObjectArray.find(
+    (currency) => currency.base === baseCurrency
+  );
+
+  selectedCurrency.rates[quoteName] = +quoteRate;
+  document.getElementById("quote-rate-beta").value = "";
+  console.log(brandObjectArray);
+
+  //Calling function: Display currencies in table
+  currencyDisplayTable();
+}
+
+function currencyDisplayTable() {
+  const gridContainer = document.getElementById(
+    "currency-rates-grid-container"
+  );
+
+  // Clear the previous content in the grid container
+  gridContainer.innerHTML = "";
+
+  // Create a table element
+  const table = document.createElement("table");
+
+  // Create header row
+  const headerRow = table.createTHead().insertRow(0);
+  headerRow.insertCell(0).textContent = "Base Currency";
+  headerRow.insertCell(1).textContent = "Quote Currency";
+  headerRow.insertCell(2).textContent = "Rate";
+
+  // Create rows for each currency object
+  brandObjectArray.forEach((currency) => {
+    for (const quoteCurrency in currency.rates) {
+      const row = table.insertRow();
+      //creating 3 cells as there are 3 currencies: EUR, USD, DKK
+      row.insertCell(0).textContent = currency.base;
+      row.insertCell(1).textContent = quoteCurrency;
+      row.insertCell(2).textContent = currency.rates[quoteCurrency];
+    }
+  });
+
+  // Append the table to the grid container
+  gridContainer.appendChild(table);
+}
+
+//--- Second part of task WEEK 2   -------------------------------------------------------------
+// Getting a range for rates, searching for rates in brandObjectArray and show filterede data.
+let fromRate;
+let toRate;
+
+document
+  .getElementById("check-if-rate-exist")
+  .addEventListener("click", checkForRateExistancy);
+
+document
+  .getElementById("check-if-rate-exist")
+  .addEventListener("reset", resetValueBoxes);
+
+function checkForRateExistancy(event) {
+  event.preventDefault();
+  fromRate = document.getElementById("rate-from").value;
+  toRate = document.getElementById("rate-to").value;
+  filteredCurrencyDisplayTable();
+}
+
+function resetValueBoxes(event) {
+  event.preventDefault();
+  document.getElementById("rate-from").value = "";
+  document.getElementById("rate-to").value = "";
+}
+
+function filteredCurrencyDisplayTable() {
+  const gridContainer = document.getElementById(
+    "filtered-currency-rates-grid-container"
+  );
+
+  // Clear the previous content in the grid container
+  gridContainer.innerHTML = "";
+
+  // Create a table element
+  const table = document.createElement("table");
+
+  // Create header row
+  const headerRow = table.createTHead().insertRow(0);
+  headerRow.insertCell(0).textContent = "Base Currency";
+  headerRow.insertCell(1).textContent = "Quote Currency";
+  headerRow.insertCell(2).textContent = "Rate";
+
+  // Creating rows for each filtered currency object
+  brandObjectArray.forEach((currency) => {
+    for (const quoteCurrency in currency.rates) {
+      if (
```

Comment:
Great Job! But I do not think you need two tables, you have already the whole logic in `currencyDisplayTable`

so lets use that one, what do you think? :)

Then the condition can look something like
```
   const shouldFilter = fromRate && toRate
      if (
        !shouldFilter ||
        (shouldFilter &&
          currency.rates[quoteCurrency] >= +fromRate &&
          currency.rates[quoteCurrency] <= +toRate)
      ) {
```

Reply:
I have made the first table display function, currencyDisplayTable, as a comment and have kept the second one. Would it be ok? :)
I will push the new changes later :)

Reply:
oki :)

Reply:
ok I missunderstood, now the user wont be able to add the new rates...